nautobot/dcim/factory.py

```diff
@@ -0,0 +1,83 @@
+import factory
+from factory.django import DjangoModelFactory
+
+from nautobot.dcim.models import Location, LocationType, Region, Site
+from nautobot.tenancy.models import Tenant
+from nautobot.utilities.factory import random_instance
+
+
+class RegionFactory(DjangoModelFactory):
+    class Meta:
+        model = Region
+        exclude = (
+            "has_parent",
+            "has_description",
+        )
+
+    name = factory.Sequence(lambda n: f"region-{n:02d}")
+
+    has_parent = factory.Faker("pybool")
+    parent = factory.Maybe("has_description", random_instance(Region), None)
+
+    has_description = factory.Faker("pybool")
+    description = factory.Maybe("has_description", factory.Faker("sentence"), "")
+
+
+class SiteFactory(DjangoModelFactory):
+    class Meta:
+        model = Site
+
+    name = factory.Sequence(lambda n: f"site-{n:02d}")
+    region = random_instance(Region)
+
+
+class LocationTypeFactory(DjangoModelFactory):
+    class Meta:
+        model = LocationType
+        exclude = (
+            "has_parent",
+            "has_description",
+        )
+
+    name = factory.Sequence(lambda n: "Root" if n == 0 else ("Building" if n == 1 else ("Floor" if n == 2 else "Room")))
+    parent = factory.Sequence(
+        lambda n: None
+        if (n == 0 or n == 1)
+        else (LocationType.objects.get(name="Building") if n == 2 else LocationType.objects.get(name="Floor"))
+    )
+
+    has_description = factory.Faker("pybool")
+    description = factory.Maybe("has_description", factory.Faker("sentence"), "")
+
```

Comment:
Since we added `Elevator` and `Aisle` LocationType, what are the appropriate contenttypes for those? or should I just set them to all contenttypes permitted?

Reply:
It's entirely arbitrary! Go with whatever feels good to you. :-)